crates/wassette/src/lib.rs

```diff
@@ -319,7 +319,43 @@ impl DownloadedResource {
                     file.file_name()
                         .context("Path to copy is missing filename")?,
                 );
-                tokio::fs::rename(file, dest).await?;
+                #[cfg(test)]
```

Comment:
I don't really love having test only code in the main code path, even if it is feature flagged. Given that this is a well documented error, I don't know that needing to add this block is worth adding the test as the logic is very clear and easily reviewable. Put another way, I don't think feature flagging this and needing to add all the code for `exdev_test_support` is worth testing whether it can handle a return value of `18`. If this were actually testing a cross device move, I'd be more open to it, but as it stands now, I don't think it is worth it.

Thoughts?

Reply:
yeah fair, I had a hard time implementing a test case for cross-device so had to simulate it here. I can remove the tests if we are confident about the logic being implemented here.

Reply:
Your logic was very clear to me fwiw